processor/src/main/kotlin/com/intuit/hooks/plugin/ksp/validation/AnnotationValidations.kt

```diff
@@ -61,12 +74,20 @@ private fun hasCodeGenerator(annotation: HookAnnotation): ValidatedNel<HookValid
     HookValidationError.NoCodeGenerator(annotation).invalidNel()
 }
 
-private fun mustBeHookType(annotation: HookAnnotation): ValidatedNel<HookValidationError, HookSignature> = try {
+private fun mustBeHookType(annotation: HookAnnotation, parentResolver: TypeParameterResolver): ValidatedNel<HookValidationError, HookSignature> = try {
+    val isSuspend: Boolean = annotation.hookFunctionSignatureType.modifiers.contains(Modifier.SUSPEND)
+    // I'm leaving this here because KSP knows that it's (String) -> Int, whereas once it gets to Poet, it's just kotlin.Function1<kotlin.Int, kotlin.String>
+    val text = annotation.hookFunctionSignatureType.text
+    val hookFunctionSignatureType = annotation.hookFunctionSignatureType.toTypeName(parentResolver)
+    val returnType = annotation.hookFunctionSignatureReference.returnType.toTypeName(parentResolver)
+    val returnTypeType = annotation.hookFunctionSignatureReference.returnType.element?.typeArguments?.firstOrNull()?.toTypeName(parentResolver)
```

Comment:
@sugarmanz all the ksp -> poet stuff is in this file now. I think this might be sufficient for this PR for right now. We can update the validation stuff to use the poet constructs in another PR.